lib/write-translations.js

```diff
@@ -137,6 +146,58 @@ function execute() {
             }
             translations['pages-strings'][text + '|' + description] = text;
           }
+
+          if (
+            path.node.type === 'CallExpression' &&
+            path.node.callee.name === 'translate'
+          ) {
+            let description = DEFAULT_DESCRIPTION;
+
+            switch (path.node.arguments.length) {
+              case 2:
+                if (path.node.arguments[1].type !== 'StringLiteral') {
+                  console.warn(
+                    `Skipping "translate" call in ${file} at line ${
+                      path.node.loc.start.line
+                    } when writing to ${FILE_TRANSLATIONS}. To enable autogeneration you must pass description as ` +
+                      `a string literal. Expressions and templates are ignored. You can still add desired strings ` +
+                      `to ${FILE_TRANSLATIONS} with your hands.`
+                  );
+
+                  break;
+                }
+                description = prepareTranslationText(
+                  path.node.arguments[1].value
+                );
+              case 1:
+                if (path.node.arguments[0].type !== 'StringLiteral') {
+                  console.warn(
```

Comment:
This `console.warn` seems to be the same as above. Please extract it out into a function.